agreement_rebate/wizards/invoice_create.py

```diff
@@ -0,0 +1,120 @@
+    domain = fields.Selection("_domain_selection", string="Domain", default="sale")
+    journal_id = fields.Many2one(
+        comodel_name="account.journal", string="Journal", required=True,
+    )
+    agreement_type_ids = fields.Many2many(
+        comodel_name="agreement.type", string="Agreement types",
+    )
+    agreement_ids = fields.Many2many(comodel_name="agreement", string="Agreements",)
+    settlements_ids = fields.Many2many(
+        comodel_name="agreement.rebate.settlement", string="Rebate settlements",
+    )
+    product_id = fields.Many2one(
+        comodel_name="product.product", string="Product", required=True,
+    )
+    invoice_group = fields.Selection(
+        [
+            ("settlement", "Settlement"),
+            ("partner", "Partner"),
+            ("commercial_partner", "Commercial partner"),
+        ],
+        required=True,
+        default="settlement",
+        string="Group invoice by",
+    )
+    invoice_partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Force invoice to"
+    )
+    invoice_type = fields.Selection(
+        [
+            ("out_invoice", "Customer Invoice"),
+            ("in_invoice", "Vendor Bill"),
+            ("out_refund", "Customer Credit Note"),
+            ("in_refund", "Vendor Credit Note"),
+        ],
+        required=True,
+        default="out_invoice",
+    )
+
+    @api.model
+    def _domain_selection(self):
+        return self.env["agreement"]._domain_selection()
+
+    def _prepare_settlement_domain(self):
+        domain = [
+            ("line_ids.rebate_type", "!=", False),
+        ]
+        if self.date_from:
+            domain.extend([("date_to", ">=", self.date_from)])
+        if self.date_to:
+            domain.extend([("date_to", "<=", self.date_to)])
+        if self.settlements_ids:
+            domain.extend([("id", "in", self.settlements_ids.ids)])
+        elif self.agreement_ids:
+            domain.extend([("line_ids.agreement_id", "in", self.agreement_ids.ids)])
+        elif self.agreement_type_ids:
+            domain.extend(
+                [
+                    (
+                        "line_ids.agreement_id.agreement_type_id",
+                        "in",
+                        self.agreement_type_ids.ids,
+                    )
+                ]
+            )
+        else:
+            domain.extend(
+                [("line_ids.agreement_id.agreement_type_id.domain", "=", self.domain)]
+            )
+        return domain
+
+    def action_create_invoice(self):
+        self.ensure_one()
+        settlements = self.env["agreement.rebate.settlement"].search(
+            self._prepare_settlement_domain()
+        )
+        settlements -= settlements.filtered(
+            lambda s: any(
+                ail.move_id.type == self.invoice_type
+                for ail in s.line_ids.mapped("invoice_line_ids").filtered(
+                    lambda ln: ln.state == "invoiced"
```

Comment:
Do you want filter settlement with state invoiced or account.move.line with parent_state (draft|post)?
state field don't exists in account.move.line